static/app/components/organizations/pageFilters/globalSelectionHeader.tsx

```diff
@@ -216,9 +209,12 @@ class GlobalSelectionHeader extends Component<Props, State> {
     callIfFunction(this.props.onUpdateEnvironments, environments);
   };
 
-  handleUpdateProjects = () => {
-    const {projects} = this.state;
+  handleChangeProjects = (projects: State['projects']) => {
+    this.setState({projects});
+    callIfFunction(this.props.onChangeProjects, projects);
+  };
```

Comment:
There's a subtle change in how the `onChange` and `onApplyChange ` handlers are being called in the FC version. Previously it was guaranteed that `onApplyChange ` would be called in the tick after `onChange`. Now they are called synchronously.

The fix is that `onApplyChange ` should now ALWAYS receive the project list, so we can count on that for updates.